### PR TITLE
Update ESLint configuration

### PR DESCRIPTION
**Pull request by taion**
Cleaning this up a bit in light of https://github.com/rackt/history/pull/112:

- Upgrade `eslint-config-airbnb` - the old version is missing some stuff like `no-undef` (the new one adds `id-length` which has to be disabled though)
- Remove redundant settings in `.eslintrc`
- Don't need to ignore `lib/` since `npm run lint` already specifies explicit included directories
- Move `mocha` environment settings into specific test/example directories rather than at top level

### CODE REVIEW

File: examples/.eslintrc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "env": {
+    "mocha": true
```
Comment (gaearon) [inline examples/.eslintrc]: Does this make sense here? Should each example have it in test folder instead?

Comment (taion) [inline examples/.eslintrc]: I can do that if you prefer. Thought this made the most sense w/r/t getting the mocha environment setting out of the top-level `.eslintrc` without adding too many sub-`.eslintrc`s.

Your call.

Comment (gaearon) [inline examples/.eslintrc]: If we split seems like we should split all the way, or it's confusing.

**Comment by taion** [comment]
I updated the `.eslintrc` files under `examples/` to only be in the respective `test` directories.

**Comment by omnidan** [comment]
@taion looks good now :ok_hand:

**Comment by mjackson** [comment]
@taion Is the plan to eventually use the rackt ESLint config?

**Comment by taion** [comment]
I don't have a plan - I just noticed that this repo was using an old version of `eslint-config-airbnb` that did not include the extremely useful `no-undef` rule and thought I'd fix it.

Probably easier to merge anyway given a smaller base of rules on top of e.g. the airbnb config.

**Pull request merged**